central/resourcecollection/datastore/datastore_impl.go

```diff
@@ -228,9 +348,32 @@ func (ds *datastoreImpl) DryRunAddCollection(ctx context.Context, collection *st
 		return err
 	}
 
+	if collection == nil {
+		return errors.New("passed collection must not be nil")
+	}
+
 	return ds.dryRunAddCollectionInGraph(collection)
 }
 
+func (ds *datastoreImpl) UpdateCollection(ctx context.Context, collection *storage.ResourceCollection) error {
+	if collection == nil {
+		return errors.New("passed collection must not be nil")
+	}
```

Comment:
factor this out into `validateCollectionIsNonEmpty(...)` and replace other use instances.